pkg/pillar/evetpm/tpm.go

```diff
@@ -625,11 +648,31 @@ func SealDiskKey(key []byte, pcrSel tpm2.PCRSelection) error {
 		return fmt.Errorf("NVWrite %v failed: %v", TpmSealedDiskPubHdl, err)
 	}
 
-	// save a snapshot of PCR values
+	// save a snapshot of current PCR values
 	if err := saveDiskKeySealingPCRs(TpmSavedDiskSealingPcrs); err != nil {
 		return fmt.Errorf("saving snapshot of sealing PCRs failed: %w", err)
 	}
 
+	// Save the previous pair of logs if any, so at most we have two pairs of
```

Comment:
I don't immediately see where the two pairs come from. From the names of the functions you call I see a single save call at the end of a single file.